src/plugins/decor/index.tsx

```diff
@@ -0,0 +1,195 @@
+            useAuthorizationStore.getState().init();
+            useCurrentUserDecorationsStore.getState().clear();
+            useUsersDecorationsStore.getState().fetch(UserStore.getCurrentUser().id, true);
+        },
+        USER_PROFILE_MODAL_OPEN: data => {
+            useUsersDecorationsStore.getState().fetch(data.userId, true);
+        },
+        MESSAGE_CREATE: data => {
+            const channelId = SelectedChannelStore.getChannelId();
+            if (data.channelId === channelId) {
+                useUsersDecorationsStore.getState().fetch(data.message.author.id);
+            }
+        },
+        TYPING_START: data => {
+            const channelId = SelectedChannelStore.getChannelId();
+            if (data.channelId === channelId) {
+                useUsersDecorationsStore.getState().fetch(data.userId);
+            }
+        },
+        LOAD_MESSAGES_SUCCESS: ({ messages }) => {
+            useUsersDecorationsStore.getState().fetchMany(messages.map(m => m.author.id));
+        },
+        // TODO: Still need to fetch for member list
+    },
+
+    set DecorationGridItem(e: any) {
+        setDecorationGridItem(e);
+    },
+
+    set DecorationGridDecoration(e: any) {
+        setDecorationGridDecoration(e);
+    },
+
+    SKU_ID,
+
+    async start() {
+        useUsersDecorationsStore.getState().fetch(UserStore.getCurrentUser().id, true);
+    },
+
+    patchUser(user) {
+        if (!user) return user;
+
+        const store = useUsersDecorationsStore.getState();
+
+        // No decor fetched yet
+        if (!store.has(user.id)) return user;
+
+        const decoration = store.get(user.id);
+
+        // Decor decoration is not null and the user doesn't have their decor decoration yet
+        if (decoration && user.avatarDecoration?.skuId !== SKU_ID) {
+            user.avatarDecoration = {
+                asset: decoration,
+                skuId: SKU_ID
+            };
+            // if the decor decoration is null, but the user has a decor decoration, remove it
+        } else if (!decoration && user.avatarDecoration && user.avatarDecoration.skuId === SKU_ID) {
+            user.avatarDecoration = null;
+        }
+
+        user.avatarDecorationData = user.avatarDecoration;
+
+        return user;
+    },
+
+    getDecorAvatarDecorationURL({ avatarDecoration, canAnimate }) {
+        // Only Decor avatar decorations have this SKU ID
+        if (avatarDecoration?.skuId === SKU_ID) {
+            const parts = avatarDecoration.asset.split("_");
+            if (!canAnimate && parts[0] === "a") parts.shift();
+            return CDN_URL + `/${parts.join("_")}.png`;
+        } else if (avatarDecoration?.skuId === RAW_SKU_ID) {
+            return avatarDecoration.asset;
+        }
+    },
+
+    DecorSection: ErrorBoundary.wrap(() => {
+        const authorization = useAuthorizationStore();
+        const { selectedDecoration, select: selectDecoration, fetch } = useCurrentUserDecorationsStore();
```

Comment:
much similar to select, giving this property a more specific name makes the code easier to read (and avoids confusing it with the builtin fetch)
```suggestion
        const { selectedDecoration, select: selectDecoration, fetch: fetchDecoration } = useCurrentUserDecorationsStore();
```